zerver/views/streams.py

```diff
@@ -296,14 +301,20 @@ def add_subscriptions_backend(
                 ])
             )),
         invite_only: bool=REQ(validator=check_bool, default=False),
-        is_announcement_only: bool=REQ(validator=check_bool, default=False),
+        is_announcement_only: Optional[bool]=REQ(validator=check_bool, default=None),
+        stream_post_policy: int=REQ(validator=check_int_in(
+            Stream.STREAM_POST_POLICY_TYPES), default=Stream.STREAM_POST_POLICY_EVERYONE),
         history_public_to_subscribers: Optional[bool]=REQ(validator=check_bool, default=None),
         announce: bool=REQ(validator=check_bool, default=False),
         principals: List[str]=REQ(validator=check_list(check_string), default=[]),
         authorization_errors_fatal: bool=REQ(validator=check_bool, default=True),
 ) -> HttpResponse:
     stream_dicts = []
     color_map = {}
+    if is_announcement_only is not None:
+        stream_post_policy = Stream.STREAM_POST_POLICY_EVERYONE  # nocoverage
+        if is_announcement_only:  # nocoverage
+            stream_post_policy = Stream.STREAM_POST_POLICY_ADMINS  # nocoverage
```

Comment:
I don't think we need backwards-compatibility here (in the stream creation interface).  Only in the events dictionaries and the "fetch some streams" API.  

(Also, we generally don't allow `nocoverage` in views code -- if we were going to support this, we'd want to add tests for the code path, but the right solution is probably to delete the `is_announcement_only` code here).

Reply:
> The mobile apps don't support creating streams right now, so that should be all we need to do for backwards-compatibility.

Regarding this line, I had a question.
In the android app it seems stream creation is possible (`All Streams` tab > `Create new stream` button).
Sorry, could you explain why we don't need backwards-compatibility here? I didn't quite understand.

Reply:
That doesn't support creating announcement_only streams, though, it's never going to post this parameter (it was just using the default).